test/test-model.js

```diff
@@ -0,0 +1,131 @@
+        sdkVersion: '2.1.1',
+        projectArtifactId: 'temp-test',
+      })
+      .on('end', done);
+  });
+
+  it('create with no options and prompts for coverage', function(done) {
+    helpers.run(path.join(__dirname, '../generators/model'))
+      // generator will create a temp directory and make sure it's empty
+      .inTmpDir(function() {
+        // we want our test to run inside the previously generated directory
+        // and we don't want it to be empty, so this is a hack for that.
+        //process.chdir(path.join(this.osTempDir, 'temp-test'));
+        process.chdir(osTempDir);
+      })
+      .on('end', done);
+  });
+
+  it('create model files in existing project', function(done) {
+    helpers.run(path.join(__dirname, '../generators/model'))
+      // generator will create a temp directory and make sure it's empty
+      .inTmpDir(function() {
+        // we want our test to run inside the previously generated directory
+        // and we don't want it to be empty, so this is a hack for that.
+        //process.chdir(path.join(this.osTempDir, 'temp-test'));
+        process.chdir(osTempDir);
+      })
+      .withOptions({
+        "model-name": "testModel",
+        "model-description": "test desc",
+        "model-author": "test author",
+        "model-version": "1.0",
+        "namespace-prefix": "zz",
+      })
+      .on('end', function() {
+        assert.file([
+          path.join(osTempDir, 'repo-amp/src/main/amp/config/alfresco/module/repo-amp/model/generated/testModel.xml'),
+          path.join(osTempDir, 'repo-amp/src/main/amp/config/alfresco/module/repo-amp/context/generated/test-model-context.xml'),
+        ]);
+        done();
+      });
+  });
+
+  it('create with invalid version', function(done) {
+    helpers.run(path.join(__dirname, '../generators/model'))
+      // generator will create a temp directory and make sure it's empty
+      .inTmpDir(function() {
+        // we want our test to run inside the previously generated directory
+        // and we don't want it to be empty, so this is a hack for that.
+        //process.chdir(path.join(this.osTempDir, 'temp-test'));
+        process.chdir(osTempDir);
+      })
+      .withOptions({
+        "model-name": "invalidVersionModel",
+        "model-description": "test desc",
+        "model-author": "test author",
+        "model-version": "asd",
+        "namespace-prefix": "zz",
+      })
+      .on('end', function() {
+        //console.log(fs.readdirSync(path.join(osTempDir, 'repo-amp/src/main/amp/config/alfresco/module/repo-amp/model/generated')));
+        assert.noFile([
+          path.join(osTempDir, 'repo-amp/src/main/amp/config/alfresco/module/repo-amp/model/generated/invalidVersionModel.xml'),
+          path.join(osTempDir, 'repo-amp/src/main/amp/config/alfresco/module/repo-amp/context/generated/invalidversion-model-context.xml'),
+        ]);
+        done();
+      });
+  });
+
+  it('handle integer version', function(done) {
```

Comment:
Can we move this code to the filter tests (it will be faster there and probably clearer what is being tested)?